cmd/internal/cli/actions_linux.go

```diff
@@ -125,6 +127,12 @@ func execStarter(cobraCmd *cobra.Command, image string, args []string, name stri
 	syscall.Umask(0022)
 
 	starter := buildcfg.LIBEXECDIR + "/singularity/bin/starter-suid"
+	// setuid workflow is not required for root user or if we
+	// are already running inside a user namespace
+	if uid == 0 || insideUserNs {
+		starter = buildcfg.LIBEXECDIR + "/singularity/bin/starter"
```

Comment:
Would it make sense here to use `filepath.Join()`?

Reply:
Yep !